### ISSUE TITLE
No exception raised when bulk api query entity mismatch 

### ISSUE THREAD

**Issue by cbailiss** [simple-salesforce/simple-salesforce#502]
I accidentally ran the following (mismatch between Account type and Contact specified in the SQL):

`fetch_generator = sf.bulk.Account.query_all("SELECT AccountId, AssistantName, AssistantPhone, Birthdate FROM Contact", lazy_operation=True)`

I was puzzled for a while since no exception was raised - I simply retrieved no records.  Is it correct that no exception was raised?  Or was I missing something such as an error value I should check?

**Comment by hkajiwara** [comment]
@cbailiss 

In that case, Salesforce is supposed to raise **_InvalidBatch_** error as follows, but it's not handled in _simple salesforce_. I would recommend you check each batch in [Monitor Bulk Data Load Jobs](https://help.salesforce.com/s/articleView?id=sf.monitoring_async_api_jobs.htm&type=5) page on your organization.

```
{
  ...
  "state": "Failed",
  "stateMessage": "InvalidBatch : Failed to process query: INVALID_TYPE_FOR_OPERATION: The root entity of the requested query (Contact) does not match the entity of the requested Bulk API Job (Account).",
  ...
}
```

**Comment by hkajiwara** [comment]
#494 should be merged into main v11.11.4.

**Comment by jon-wobken** [comment]
resolved from merge #494
